packages/core/src/messagingProtocols/mls/EventHandler/events/welcomeMessage/welcomeMessage.ts

```diff
@@ -35,6 +35,9 @@ const handleWelcomeMessage = async ({mlsService, event}: HandleWelcomeMessagePar
   const newGroupId = await mlsService.processWelcomeMessage(data);
   const groupIdStr = Encoder.toBase64(newGroupId).asString;
   // The groupId can then be sent back to the consumer
+
+  // After we were added to the group we need to schedule a periodic key material renewal
+  mlsService.scheduleKeyMaterialRenewal(groupIdStr);
```

Comment:
We were not handling this case previously, we should also maintain the timer when being added to a group (not only when creating one) - basically we should maintain the timer for all the groups we're currently in (the way we've joined it does not matter)